From ef6d749c0e2dd53719046c843796f5e142c43679 Mon Sep 17 00:00:00 2001
From: Peter Soetens <peter@thesourceworks.com>
Date: Sat, 16 Feb 2013 01:21:43 +0100
Subject: [PATCH] scripting: fix crash in quantal and possibly all other OS'es.

Parser rules on stack got cleaned up before they could be used.

Signed-off-by: Peter Soetens <peter@thesourceworks.com>
---
 rtt/scripting/CommonParser.cpp |    1 +
 rtt/scripting/ScriptParser.cpp |    8 ++++++--
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/rtt/scripting/CommonParser.cpp b/rtt/scripting/CommonParser.cpp
index 95f059b..a5f2753 100644
--- rtt/scripting/CommonParser.cpp
+++ rtt/scripting/CommonParser.cpp
@@ -106,6 +106,7 @@ namespace RTT {
         BOOST_SPIRIT_DEBUG_RULE( lexeme_identifier );
         BOOST_SPIRIT_DEBUG_RULE( lexeme_notassertingidentifier );
         BOOST_SPIRIT_DEBUG_RULE( type_name );
+        BOOST_SPIRIT_DEBUG_RULE( skipper );
 
         // an identifier is a word which can be used to identify a
         // label, or be the name of an object or method.  it is required
diff --git a/rtt/scripting/ScriptParser.cpp b/rtt/scripting/ScriptParser.cpp
index bfd3af1..5fb3d18 100644
--- rtt/scripting/ScriptParser.cpp
+++ rtt/scripting/ScriptParser.cpp
@@ -162,8 +162,12 @@ namespace RTT
         //skip_parser_t skip_parser = SKIP_PARSER;
         //iter_pol_t iter_policy( skip_parser );
         //#define SKIP_PARSER
-        iter_pol_t iter_policy((comment_p("#") | comment_p("//") | comment_p(
-                "/*", "*/") | (space_p - eol_p) | commonparser->skipper));
+        skip_parser_t skippers = (comment_p("#") | comment_p("//") 
+                          | comment_p("/*", "*/") 
+                          | (space_p - eol_p) 
+			 | (commonparser->skipper));
+
+        iter_pol_t iter_policy(skippers);
         scanner_pol_t policies(iter_policy);
         scanner_t scanner(begin, end, policies);
 
-- 
1.7.1

